daemons/controld/controld_join_dc.c

```diff
@@ -636,8 +654,14 @@ finalize_join_for(gpointer key, gpointer value, gpointer user_data)
     fsa_cib_anon_update(XML_CIB_TAG_NODES, tmp1);
     free_xml(tmp1);
 
+    if (join_node->join == crm_join_nack_quiet) {
+        crm_trace("Not sending nack message to node %s with feature set older "
+                  "than 3.17.0", join_to);
+        return;
```

Comment:
The next block (for inactive peers) sets the expected state to pending. I don't think that's appropriate here but want to make a note of it.

I'm assuming this should come after whatever the `fsa_cib_anon_update` is doing (previous block), but to be honest I haven't dived into that.

Reply:
The previous block is updating the \<node\> element (not node_state as the comment suggests) with the node's uname and uuid, in case they weren't known before. "anon_update" is named for using the daemon user rather than a specific username, but the controller never uses a specific username, so the difference in practice is it's fire-and-forget rather than getting a call ID for registering a callback.